### PR TITLE
Adds deepspeech stt

### PR DESCRIPTION
**Pull request by pietrop**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      

_NA_

**Describe what the PR does**    

Adds support for DeepSpeech STT

**State whether the PR is ready for review or whether it needs extra work**    

Work in progress

- [x] travis github release deploy for pietrop repo 
- [x] add `deepspeech-node-wrapper`
- [x] add logic to download model under settings in `deepspeech-node-wrapper`
- [x] add logic to download model under settings in electron app
- [x] package/deploy and test 

**Additional context**    

After removing a whole bunch of not needed files, via `.npmignore` and `.gitignore` package size is down to `24.9 kB` 🎉 

for packaged version to test locally see
https://github.com/pietrop/digital-paper-edit-electron/releases/tag/1.0.9

- [ ] raise PR against BBC repo before merging to master in this repo(?)

nice to have
- [ ] add a way to show progress of download in the UI
- [ ] Nice to have for future PRs is to add options for models for other languages eg [github.com/MozillaItalia/DeepSpeech-Italian-Model](https://github.com/MozillaItalia/DeepSpeech-Italian-Model/releases) & [github.com/mozilla/DeepSpeech/blob/master/TRAINING.rst](https://github.com/mozilla/DeepSpeech/blob/master/TRAINING.rst)

### CODE REVIEW

File: src/stt-settings/credentials.js [hunk 1]
```diff
@@ -1,28 +1,93 @@
...
-const appUserDataPath = electron.remote.app.getPath('userData');
+const fs = require("fs");
+const path = require("path");
+const electron = require("electron");
+const downloadDeepSpeechModel = require("deepspeech-node-wrapper")
+  .downloadDeepSpeechModel;
+const appUserDataPath = electron.remote.app.getPath("userData");
+// TODO: consider moving deepspeech logic to a separate file from credentials.js?
+
+function getDeepSpeechModelFolderName( modelVersion) {
```
Comment (emettely) [inline src/stt-settings/credentials.js]:
```suggestion
function getDeepSpeechModelFolderName( modelVersion = "0.6.0") {
```

One way to get default params in easily

File: src/stt-settings/credentials.js [hunk 2]
```diff
@@ -1,28 +1,93 @@
...
+}
+
+function getDeepSpeechModelPath(deepspeechModelVersion) {
+  return path.join(appUserDataPath, getDeepSpeechModelFolderName(deepspeechModelVersion));
+}
+
+// TODO: add some way to check if model,
+// folder and necessary files,
+// are set in user folder in application libary path
+function getIsDeepspeechModelSet() {
```
Comment (emettely) [inline src/stt-settings/credentials.js]: Would probably be good if here or somewhere else there's a link to what each of the files and folders are and why they are necessary for the deepspeech e.g. this link https://github.com/pietrop/deepspeech-node-wrapper

File: docs/ADR/13-12-2019-deepspeech-stt-electron-integration-adr.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Deepspeech STT Electron Integration ADR
+
+adding support for Mozilla DeepSpeech, for higher quality STT for offline recognition.
+[`deepspeech-node-wrapper`](https://github.com/pietrop/deepspeech-node-wrapper)
+
+Problem as discussed in [ADR in that repository](https://github.com/pietrop/deepspeech-node-wrapper/blob/master/docs/ADR/2019-12-10-deepspeech-stt.md) is how to package the model, as it's size is 1.8gb
+
+previously open source offline STT integrated with electron [pocketsphinx](https://github.com/OpenNewsLabs/pocketsphinx-stt) where model is self contained in npm, as it's not very bick. And Gentle, that [requires spinning up a separate app](https://autoedit.gitbook.io/user-manual), eg as used in [autoEdit](http://www.autoedit.io/).
```
Comment (emettely) [inline docs/ADR/13-12-2019-deepspeech-stt-electron-integration-adr.md]:
```suggestion
previously open source offline STT integrated with electron [pocketsphinx](https://github.com/OpenNewsLabs/pocketsphinx-stt) where model is self contained in npm, as it's not very big (40MB). And Gentle, that [requires spinning up a separate app](https://autoedit.gitbook.io/user-manual), eg as used in [autoEdit](http://www.autoedit.io/).
```

File: src/stt-settings/credentials.js [hunk 4]
```diff
@@ -1,28 +1,93 @@
...
+const fs = require("fs");
+const path = require("path");
+const electron = require("electron");
+const downloadDeepSpeechModel = require("deepspeech-node-wrapper")
+  .downloadDeepSpeechModel;
+const appUserDataPath = electron.remote.app.getPath("userData");
+// TODO: consider moving deepspeech logic to a separate file from credentials.js?
+
+function getDeepSpeechModelFolderName( modelVersion) {
+  const deepspeechModelVersion = modelVersion ? modelVersion : "0.6.0";
```
Comment (emettely) [inline src/stt-settings/credentials.js]:
```suggestion
```

File: src/stt-settings/credentials.js [hunk 5]
```diff
@@ -1,28 +1,93 @@
...
+  return `deepspeech-${deepspeechModelVersion}-models`;
```
Comment (emettely) [inline src/stt-settings/credentials.js]:
```suggestion
  return `deepspeech-${modelVersion}-models`;
```

File: src/stt-settings/credentials.js [hunk 6]
```diff
@@ -1,28 +1,93 @@
...
+  // return deepspeeshmodels
+}
+
+function getDeepSpeechModelPath(deepspeechModelVersion) {
+  return path.join(appUserDataPath, getDeepSpeechModelFolderName(deepspeechModelVersion));
+}
+
+// TODO: add some way to check if model,
+// folder and necessary files,
+// are set in user folder in application libary path
```
Comment (emettely) [inline src/stt-settings/credentials.js]:
```suggestion
// are set in user folder in application libary path
// Files required described in README of https://github.com/pietrop/deepspeech-node-wrapper
```
^ it's mentioned but not described IIRC

**Comment by pietrop** [comment]
see https://github.com/bbc/digital-paper-edit-electron/pull/11

**Pull request merged**